src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -1,197 +1,599 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+
 namespace System
 {
+    /// <summary>
+    /// Represents a pseudo-random number generator, which is an algorithm that produces a sequence of numbers
+    /// that meet certain statistical requirements for randomness.
+    /// </summary>
     public class Random
     {
-        private static readonly Random s_globalRandom = new Random(GenerateGlobalSeed());
-        [ThreadStatic]
-        private static Random? t_threadRandom;
+        /// <summary>The underlying generator implementation.</summary>
+        /// <remarks>
+        /// This is separated out so that different generators can be used based on how this Random instance is constructed.
+        /// If it's built from a seed, then we may need to ensure backwards compatibility for folks expecting consistent sequences
+        /// based on that seed.  If the instance is actually derived from Random, then we need to ensure the derived type's
+        /// overrides are called anywhere they were being called previously.  But if the instance is the base type and is constructed
+        /// with the default constructor, we have a lot of flexibility as to how to optimize the performance and quality of the generator.
+        /// </remarks>
+        private readonly ImplBase _impl;
+
+        /// <summary>Initializes a new instance of the <see cref="Random"/> class using a default seed value.</summary>
+        public Random() =>
+            // With no seed specified, if this is the base type, we can implement this however we like.
+            // If it's a derived type, for compat we respect the previous implementation.
+            _impl = GetType() == typeof(Random) ?
```

Comment:
Do you think it's worthwhile to have an overload letting derived users opt-in to the new algorithm?
Likewise, were there any considerations about an `AppContext` switch for more global opt-in?

Reply:
It would tie our hands wrt changing or improving the algorithm again. I do not think it is worthwhile.